### ISSUE TITLE
need better documentation

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaCloud/AWSS3.jl#198]
Currently this still links to docs from the deprecated `AWSCore.jl`.  I think it would be nice if this package had its own documenter.jl setup, we can of course provide links to AWS.jl docs.  I can do a PR for this after my current one gets merged, but before I do, was there any plan here?  Didn't want to do this if you guys already planned on putting docs in AWS.jl or something.

**Comment by ericphanson** [comment]
I'm not aware of any plan but I'd love for AWSS3 to have some dedicated docs!

**Comment by ExpandingMan** [comment]
Ok, cool, I'll definitely do this.

It's kind of amazing and a little scary how important a package this so often winds up being.

**Comment by mattBrzezinski** [comment]
Oh good catch! Personally I never use the self-hosted documentation. But I know a lot of other people do. Having them for this package would be nice!

**Comment by ExpandingMan** [comment]
The main, and very important, benefit of the Documenter.jl docs over just using a README are the API docs.  These tend to be the most important part of the documentation, yet they're a piece of documentation that is almost always missing from README.

**Comment by mattBrzezinski** [comment]
> The main, and very important, benefit of the Documenter.jl docs over just using a README are the API docs. These tend to be the most important part of the documentation, yet they're a piece of documentation that is almost always missing from README.

Yeah I just always end up looking at the source code to figure stuff out. But I'm definitely the minority here.

**Comment by ExpandingMan** [comment]
I'm not opposed to that technique, but the thing is, if people don't have API docs they tend not to put (good) doc strings in the code.  It's kind of a catch 22 :laughing:

**Comment by mattBrzezinski** [comment]
> I'm not opposed to that technique, but the thing is, if people don't have API docs they tend not to put (good) doc strings in the code. It's kind of a catch 22 😆

Definitely, I'm very much in favour of this issue!
